### ISSUE TITLE
strict-type-predicates and assertNever

### ISSUE THREAD

**Issue by roginvs** [palantir/tslint#4217]
### Bug Report
Tslint 5.11.0, typescript 3.0.3, running via cli

#### tslint.conf
```json
{
  "rules": {
    "strict-type-predicates": true
  }
}
```
#### TypeScript code being linted

```ts
function assertNever(x: never): never {
  throw new Error(`Unexpected object: ${x}`);
}

type X = "value" | undefined;
declare const x: X;

// handy way to ensure that all union cases are checked
const y =
  x === "value"
    ? "Some value"
    : x === undefined
      ? "It is undefined"
      : assertNever(x);
```

### Workaround
```ts
// Works fine if check undefined first
const y =
  x === undefined
    ? "It is undefined"
    : x === "value"
      ? "Some value"
      : assertNever(x);
```

I suppose it should skip "always true" check if "else" branch have "never" type

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
